### ISSUE TITLE
Release code and curated data for Guiding MLLMs on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [rgonzalezp/guiding-multimodal-large-language-models-with-blind-and-low-vision-people-visual-questions#1]
Hi @rgonzalezp 🤗

Niels here from the open-source team at Hugging Face. I discovered your work on Arxiv and was wondering whether you would like to submit it to [hf.co/papers](https://hf.co/papers) to improve its discoverability. If you are one of the authors, you can submit it at https://huggingface.co/papers/submit.

The paper page lets people discuss about your paper and lets them find artifacts about it (your models, datasets or demo for instance), you can also claim
the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

I saw in your abstract that you plan to release your "paper reviews, and data analysis are publicly available in a Github repository". It would be great to make the code for your system (for guiding MLLMs) and the curated datasets (like the context set and test set derived from VizWiz-LF) available on the 🤗 hub *when they are ready*, to improve their discoverability/visibility.
We can add tags so that people find them when filtering https://huggingface.co/models and https://huggingface.co/datasets.

## Uploading code (e.g., system for guiding MLLMs)

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, if your system involves any components that could be treated as models (e.g., a specific ranking model, or parts of the guidance mechanism), we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverage the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download specific code files from the hub.

We encourage researchers to push each distinct component or model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the code/checkpoints to the paper page.

## Uploading curated dataset

Would be awesome to make the curated context set and test set available on 🤗 , so that people can do:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/your-dataset")
```
See here for a guide: https://huggingface.co/docs/datasets/loading.

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the first few rows of the data in the browser.

Let me know if you're interested/need any help regarding this!

Cheers,

Niels
ML Engineer @ HF 🤗

**Comment by rgonzalezp** [comment]
Hey Niels,

This sounds like a very good idea. We are in the process of preparing our
supplementary material to upload to github.

I think it would be interesting to also upload some of our material to
hugging face based on your suggestion (probably the curated dataset).

Do you have an example of a paper uploaded to hugging face with good
discoverability as an example to follow?

*Ricky Gonzalez Penuela*
*Information Science PhD Candidate*
*Cornell Tech, Cornell University*

**Comment by NielsRogge** [comment]
Hi @rgonzalezp thanks for your reply! Here's a good example of a recent VLM that published everything on the hub: https://huggingface.co/papers/2509.23661.
